From 14bda7ab1f9baaa567253b862ebcc7350a3f43be Mon Sep 17 00:00:00 2001
From: Aristeu Rozanski <aris@redhat.com>
Date: Thu, 10 Apr 2014 19:29:25 -0400
Subject: [edac] sb_edac: use "event" instead of "exception" when MC wasnt signaled

Message-id: <20140410192925.047989582@napanee.usersys.redhat.com>
Patchwork-id: 78860
O-Subject: [RHEL7.1 PATCH v2 1/2] sb_edac: use "event" instead of "exception" when MC wasnt signaled
Bugzilla: 1056785
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1056785
brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7325861
testing: by me in a beaker system
upstream: cf40f80cbe374e98283b9865c1295fc74a68fa29

 commit cf40f80cbe374e98283b9865c1295fc74a68fa29
 Author: Aristeu Rozanski <arozansk@redhat.com>
 Date:   Tue Mar 11 15:45:41 2014 -0400

     sb_edac: use "event" instead of "exception" when MC wasnt signaled

     Corrected Errors are MC events, not exceptions and reporting as the
     later might confuse users.

     Signed-off-by: Aristeu Rozanski <arozansk@redhat.com>
     Signed-off-by: Mauro Carvalho Chehab <m.chehab@samsung.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/edac/sb_edac.c b/drivers/edac/sb_edac.c
index a114128..a32fed9 100644
--- a/drivers/edac/sb_edac.c
+++ b/drivers/edac/sb_edac.c
@@ -1828,6 +1828,7 @@ static int sbridge_mce_check_error(struct notifier_block *nb, unsigned long val,
  struct mce *mce = (struct mce *)data;
  struct mem_ctl_info *mci;
  struct sbridge_pvt *pvt;
+ char *type;
 
  mci = get_mci_for_node_id(mce->socketid);
  if (!mci)
@@ -1843,10 +1844,15 @@ static int sbridge_mce_check_error(struct notifier_block *nb, unsigned long val,
  if ((mce->status & 0xefff) >> 7 != 1)
   return NOTIFY_DONE;
 
+ if (mce->mcgstatus & MCG_STATUS_MCIP)
+  type = "Exception";
+ else
+  type = "Event";
+
  printk("sbridge: HANDLING MCE MEMORY ERROR\n");
 
- printk("CPU %d: Machine Check Exception: %Lx Bank %d: %016Lx\n",
-        mce->extcpu, mce->mcgstatus, mce->bank, mce->status);
+ printk("CPU %d: Machine Check %s: %Lx Bank %d: %016Lx\n",
+        mce->extcpu, type, mce->mcgstatus, mce->bank, mce->status);
  printk("TSC %llx ", mce->tsc);
  printk("ADDR %llx ", mce->addr);
  printk("MISC %llx ", mce->misc);
-- 
1.7.1